### PR TITLE
GEODE-8419: SSL/TLS protocol and cipher suite configuration is ignored

### PR DESCRIPTION
**Pull request by bschuchardt**
Configure cipher suites when creating an SSLEngine

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java [hunk 1]
```diff
@@ -98,6 +102,30 @@ private void testBindExceptionMessageFormatting(InetAddress inetAddress) throws
...
+
+    SSLEngine engine = mock(SSLEngine.class);
+    when(engine.getSSLParameters()).thenReturn(parameters);
+
+    engine.setEnableSessionCreation(true);
+    socketCreator.configureSSLEngine(engine, "somehost", 12345, true);
+
+    verify(engine).setUseClientMode(isA(Boolean.class));
+    verify(engine).setSSLParameters(parameters);
+    verify(engine).setEnabledCipherSuites(isA(String[].class));
```
Comment (pivotal-jbarrett) [inline geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java:124]: Should we verify that it is called with a `String[]` only containing the `"someCipher"`?

File: geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java [hunk 2]
```diff
@@ -98,6 +102,30 @@ private void testBindExceptionMessageFormatting(InetAddress inetAddress) throws
     }
   }
 
+  @Test
+  public void configureSSLEngine() {
+    SSLConfig config = new SSLConfig.Builder().setCiphers("someCipher").setEnabled(true)
```
Comment (pivotal-jbarrett) [inline geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java:107]: What about a test for `"any"`?

File: geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java (hunk 1 above)
Comment (bschuchardt) [reply geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java:139]: yes, I was being lazy there.  I've corrected that

File: geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java (hunk 2 above)
Comment (bschuchardt) [reply geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java:111]: Yes!  I'm having lots of "fun" with "any" this week and that's an excellent idea.

File: geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java [hunk 3]
```diff
@@ -98,6 +106,65 @@ private void testBindExceptionMessageFormatting(InetAddress inetAddress) throws
...
+
+    final Object[] setCiphers = new Object[1];
+    doAnswer((Answer<Void>) invocation -> {
+      setCiphers[0] = invocation.getArgument(0);
+      return null;
+    }).when(engine).setEnabledCipherSuites(any(String[].class));
+
+    socketCreator.configureSSLEngine(engine, "somehost", 12345, true);
+
+    verify(engine).setUseClientMode(isA(Boolean.class));
```
Comment (pivotal-jbarrett) [inline geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java:137]: Are you familiar with `ArgumentCaptors` in Mokito? I find it to be more readable approach to the argument validation than `doAnswer` approach. 
https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#verify-T-

Comment (bschuchardt) [reply geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java:124]: I was not familiar with ArgumentCaptor.  Much simpler.

File: geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java [hunk 4]
```diff
@@ -120,28 +117,14 @@ public void configureSSLEngine() {
...
     verify(engine).setUseClientMode(isA(Boolean.class));
     verify(engine).setSSLParameters(parameters);
-    verify(engine).setEnabledCipherSuites(isA(String[].class));
-    assertThat(setProtocols[0]).isNotNull();
-    String[] protocolStrings = (String[]) setProtocols[0];
-    assertThat(protocolStrings[0].equals("someProtocol"));
-    String[] cipherStrings = (String[]) setCiphers[0];
-    assertThat(cipherStrings[0].equals("someCipher"));
+    verify(engine).setEnabledProtocols(engineArgumentCaptor.capture());
+    verify(engine).setEnabledCipherSuites(engineArgumentCaptor.capture());
```
Comment (pivotal-jbarrett) [inline geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java:127]: You will want separate captors for each of these and then assert the values.
```java
ArgumentCaptor<String[]> enabledProtocols = ArgumentCaptor.forClass(String[].class);
verify(engine).setEnabledProtocols(enabledProtocols.capture());
assertThat(enabledProtocols.getValue()).containsExactly("someProtocol");

ArgumentCaptor<String[]> enabledCipherSuites = ArgumentCaptor.forClass(String[].class);
verify(engine). setEnabledCipherSuites(enabledCipherSuites.capture());
assertThat(enabledCipherSuites.getValue()).containsExactly("someCipher");
```

Comment (bschuchardt) [reply geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java:127]: The captor in this commit does capture both String[] values though.  The first capture() returns the String[] with "someProtocol" and the second returns the String[] with "someCipher".  I'll add the content checks.

**Pull request merged**